tests/test_tasker.py

```diff
@@ -403,3 +404,56 @@ def test_retry_method(retry_times):
             retry(my_func, *my_args, retry=retry_times, **my_kwargs)
     else:
         retry(my_func, *my_args, retry=retry_times, **my_kwargs)
+
+
+@pytest.mark.parametrize('exc', [ProtocolError, APIError, False])
+@pytest.mark.parametrize('in_init', [True, False])
+@pytest.mark.parametrize('retry_times', [-1, 0, 1, 2, 3])
+def test_retry_generator(exc, in_init, retry_times):
+    def simplegen():
+        yield "log line"
+
+    my_args = ('some', 'new')
+    if not in_init:
+        my_kwargs = {'one': 'first', 'two': 'second', 'retry_times': retry_times}
+    else:
+        my_kwargs = {'one': 'first', 'two': 'second'}
+
+    if in_init:
+        t = DockerTasker(retry_times=retry_times)
+    else:
+        t = DockerTasker()
+
+    (flexmock(time)
+        .should_receive('sleep')
+        .and_return(None))
+
+    if not exc:
+        cr = CommandResult()
+        cr._error = "error"
+        cr._error_detail = "error_detail"
+
+    if exc == APIError:
+        response = flexmock(content="abc", status_code=408)
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_raise(APIError, "error", response))
+    elif exc == ProtocolError:
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_raise(ProtocolError, "error"))
+    else:
+        (flexmock(atomic_reactor.util)
+            .should_receive('wait_for_command')
+            .times(retry_times + 1)
+            .and_return(cr))
+
+    if retry_times >= 0:
+        with pytest.raises(RetryGeneratorException):
```

Comment:
Maybe use `pytest.raises(..., message=...):` to make sure the exception description matches what we expect?